kroxylicious-api/src/main/java/io/kroxylicious/proxy/plugin/PluginConfig.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.plugin;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation that identifies a plugin instance name at a plugin point within the configuration.
+ * This should be applied to the property of class representing the plugin point, and point to
+ * the name of the sibling property (annotated with {@link PluginReference @PluginReference}) that names the plugin instance.
+ */
```

Comment:
I'm struggling to understand what this javadoc is trying to tell me. 

I think it needs an example or some other context for it to be consumed by others.

Reply:
I agree. But it's tricky to provide an example for filter authors that isn't based on internal classes (like `FilterDefinition`). 

I guess I can provide a full example (using an exemplar plugin) in the Javadoc for one of the annotations and then link from the others?